File: examples/csharp/basic.cs
/*
 * UTM Coordinate Converter API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the UTM Coordinate Converter API.
 * API Documentation: https://docs.apiverve.com/ref/utmconverter
 */

using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using System.Linq;

namespace APIVerve.Examples
{
    class UTMCoordinateConverterExample
    {
        private static readonly string API_KEY = Environment.GetEnvironmentVariable("APIVERVE_API_KEY") ?? "YOUR_API_KEY_HERE";
        private static readonly string API_URL = "https://api.apiverve.com/v1/utmconverter";

        /// <summary>
        /// Make a GET request to the UTM Coordinate Converter API
        /// </summary>
        static async Task<JsonDocument> CallUTMCoordinateConverterAPI()
        {
            try
            {
                using var client = new HttpClient();
                client.DefaultRequestHeaders.Add("x-api-key", API_KEY);

                // Query parameters
                var queryParams &#x3D; new Dictionary&lt;string, string&gt; { [&quot;direction&quot;] &#x3D; &quot;to_utm&quot;, [&quot;latitude&quot;] &#x3D; 40.7128, [&quot;longitude&quot;] &#x3D; -74.0060, [&quot;zone&quot;] &#x3D; 18, [&quot;easting&quot;] &#x3D; 585628, [&quot;northing&quot;] &#x3D; 4511322, [&quot;hemisphere&quot;] &#x3D; &quot;N&quot; };

                var queryString = string.Join("&",
                    queryParams.Select(kvp => $"{kvp.Key}={Uri.EscapeDataString(kvp.Value)}"));
                var url = $"{API_URL}?{queryString}";

                var response = await client.GetAsync(url);

                // Check if response is successful
                response.EnsureSuccessStatusCode();

                var responseBody = await response.Content.ReadAsStringAsync();
                var data = JsonDocument.Parse(responseBody);

                // Check API response status
                if (data.RootElement.GetProperty("status").GetString() == "ok")
                {
                    Console.WriteLine("✓ Success!");
                    Console.WriteLine("Response data: " + data.RootElement.GetProperty("data"));
                    return data;
                }
                else
                {
                    var error = data.RootElement.TryGetProperty("error", out var errorProp)
                        ? errorProp.GetString()
                        : "Unknown error";
                    Console.WriteLine($"✗ API Error: {error}");
                    return null;
                }
            }
            catch (HttpRequestException e)
            {
                Console.WriteLine($"✗ Request failed: {e.Message}");
                return null;
            }
        }

        static async Task Main(string[] args)
        {
            Console.WriteLine("📤 Calling UTM Coordinate Converter API...\n");

            var result = await CallUTMCoordinateConverterAPI();

            if (result != null)
            {
                Console.WriteLine("\n📊 Final Result:");
                Console.WriteLine(result.RootElement.GetProperty("data"));
            }
            else
            {
                Console.WriteLine("\n✗ API call failed");
            }
        }
    }
}

File: examples/javascript/basic.js
/**
 * UTM Coordinate Converter API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the UTM Coordinate Converter API.
 * API Documentation: https://docs.apiverve.com/ref/utmconverter
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/utmconverter';

/**
 * Make a GET request to the UTM Coordinate Converter API
 */
async function callUTMCoordinateConverterAPI() {
  try {
    // Query parameters
    const params &#x3D; new URLSearchParams({
            direction: &#x27;to_utm&#x27;,
            latitude: 40.7128,
            longitude: -74.0060,
            zone: 18,
            easting: 585628,
            northing: 4511322,
            hemisphere: &#x27;N&#x27;
        });

    const response = await fetch(`${API_URL}?${params}`, {
      method: 'GET',
      headers: {
        'x-api-key': API_KEY
      }
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callUTMCoordinateConverterAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
